design-proposals/klip-43-timestamp-data-type-support.md

```diff
@@ -0,0 +1,159 @@
+# KLIP-43: TIMESTAMP Data Type Support
+
+**Author**: @jzaralim | 
+**Release Target**: 0.15 | 
+**Status**: In Discussion | 
+**Discussion**: https://github.com/confluentinc/ksql/pull/6649
+
+**tl;dr:** _KSQL currently converts time data into long values. This KLIP introduces a TIMESTAMP
+data type to handle time data instead._
+
+## Motivation and background
+
+With Connect integration, KSQL processes data from various sources and sinks. Most other databases
+have a TIMESTAMP or DATETIME type, which KSQL currently converts into long values. This sometimes
+causes problems when sinking into a database. Supporting TIMESTAMP types would make moving time data
+between KSQL and other data sources/sinks smoother and less error-prone.
+
+## What is in scope
+### Required
```

Comment:
What about:
  - Should we support arithmetic functions too? casting? any other basic feature should be listed here.
  - Should you support STRUCT, ARRAY, MAP as well?